cpp/src/arrow/sparse_tensor.cc

```diff
@@ -303,113 +371,4 @@ bool SparseTensor::Equals(const SparseTensor& other) const {
-                       tensor.dim_names_) {
-  SparseTensorConverter<TYPE, SparseIndexType> converter(tensor);
-  Status s = converter.Convert();
-  DCHECK_OK(s);
-  sparse_index_ = converter.sparse_index;
-  data_ = converter.data;
-}
-
-// Constructor with a dense tensor
-template <typename SparseIndexType>
-SparseTensorImpl<SparseIndexType>::SparseTensorImpl(const Tensor& tensor)
-    : SparseTensorImpl(nullptr, tensor.type(), nullptr, tensor.shape(),
-                       tensor.dim_names_) {
-  switch (tensor.type()->id()) {
-    case Type::UINT8:
-      MakeSparseTensorFromTensor<UInt8Type, SparseIndexType>(tensor, &sparse_index_,
-                                                             &data_);
-      return;
-    case Type::INT8:
-      MakeSparseTensorFromTensor<Int8Type, SparseIndexType>(tensor, &sparse_index_,
-                                                            &data_);
-      return;
-    case Type::UINT16:
-      MakeSparseTensorFromTensor<UInt16Type, SparseIndexType>(tensor, &sparse_index_,
-                                                              &data_);
-      return;
-    case Type::INT16:
-      MakeSparseTensorFromTensor<Int16Type, SparseIndexType>(tensor, &sparse_index_,
-                                                             &data_);
-      return;
-    case Type::UINT32:
-      MakeSparseTensorFromTensor<UInt32Type, SparseIndexType>(tensor, &sparse_index_,
-                                                              &data_);
-      return;
-    case Type::INT32:
-      MakeSparseTensorFromTensor<Int32Type, SparseIndexType>(tensor, &sparse_index_,
-                                                             &data_);
-      return;
-    case Type::UINT64:
-      MakeSparseTensorFromTensor<UInt64Type, SparseIndexType>(tensor, &sparse_index_,
-                                                              &data_);
-      return;
-    case Type::INT64:
-      MakeSparseTensorFromTensor<Int64Type, SparseIndexType>(tensor, &sparse_index_,
-                                                             &data_);
-      return;
-    case Type::HALF_FLOAT:
-      MakeSparseTensorFromTensor<HalfFloatType, SparseIndexType>(tensor, &sparse_index_,
-                                                                 &data_);
-      return;
-    case Type::FLOAT:
-      MakeSparseTensorFromTensor<FloatType, SparseIndexType>(tensor, &sparse_index_,
-                                                             &data_);
-      return;
-    case Type::DOUBLE:
-      MakeSparseTensorFromTensor<DoubleType, SparseIndexType>(tensor, &sparse_index_,
-                                                              &data_);
-      return;
-    default:
-      break;
-  }
-}
-
-// ----------------------------------------------------------------------
-// Instantiate templates
-
-#define INSTANTIATE_SPARSE_TENSOR(IndexType)                           \
-  template class ARROW_TEMPLATE_EXPORT SparseTensorImpl<IndexType>;    \
```

Comment:
Could you also remove `ARROW_TEMPLATE_EXPORT` definition in `cpp/src/arrow/util/visibility.h`?

Reply:
done